yapapi/rest/market.py

```diff
@@ -14,13 +15,29 @@
 class AgreementDetails(object):
     raw_details: models.Agreement
 
-    def __init__(self, *, _ref: models.Agreement):
-        self.raw_details = _ref
+    @dataclass
+    class _View:
+        """A certain fragment of an agreement's properties."""
+        properties: dict
+
+        def extract(self, m: Type[_ModelType]) -> _ModelType:
+            """Extract properties for the given model from this view's properties."""
+            return m.from_properties(self.properties)
 
-    # TODO 0.4+ update the name to something more readable
-    def view_prov(self, c: Type[_ModelType]) -> _ModelType:
+    @property
+    def provider_view(self) -> _View:
```

Comment:
I don't think a non-public class `AgreementDetails._View` should occur in the signature of public methods `AgreementDetails.provider_view/requestor_view`. Is there a point in letting library users create `_View` objects?